src/app/access-requests/features/access-request-field-edit/access-request-field-edit.component.ts

```diff
@@ -38,58 +47,249 @@ import { ConfigService } from '@app/shared/services/config.service';
 
-  rows = computed<number>(() => (this.name().includes('note') ? 5 : 1));
+  rows = computed<number[]>(() =>
+    this.names().map((x) => (x.includes('note') ? 5 : 1)),
+  );
   locked = computed<boolean>(() => this.request().status !== 'pending');
 
-  field = model<string>('');
+  fields = model<(string | Date)[]>([]);
 
-  saved = output<[keyof AccessRequest, string]>();
-  edited = output<[keyof AccessRequest, boolean]>();
+  saved = output<Map<keyof AccessRequest, string>>();
+  edited = output<Map<keyof AccessRequest, boolean>>();
 
   isOpen = signal<boolean>(false);
   isModified = signal<boolean>(false);
 
-  ticketUrl = computed<string | null>(() =>
-    this.name() == 'ticket_id' && this.field()
-      ? this.#baseTicketUrl + this.field()
-      : null,
+  ticketUrl = computed<(string | null)[]>(() =>
+    this.names().map((x, idx) =>
+      x == 'ticket_id' && this.fields()[idx]
+        ? this.#baseTicketUrl + this.fields()[idx]
+        : null,
+    ),
   );
 
-  #defaultValue = computed<string>(() => (this.request() as any)[this.name()] || '');
+  todayMidnight = new Date();
+  minFromDate = new Date();
+  minUntilDate = new Date();
+  maxFromDate = new Date();
+  maxUntilDate = new Date();
+  fieldErrorMessages = signal<string[]>([]);
+
+  #defaultValues = computed<string[]>(() =>
+    this.names().map((x) => this.request()[x] || ''),
+  );
+
+  durationFormControl: (FormControl | null)[] = [];
+
+  constructor() {
+    this.todayMidnight.setHours(0, 0, 0, 0);
+    let d = new Date();
+    d.setDate(d.getDate() + this.#config.defaultAccessDurationDays);
+    this.minUntilDate.setDate(
+      this.minUntilDate.getDate() + this.#config.accessGrantMinDays,
+    );
+    this.updateAccessEndRanges(new Date());
+    this.updateAccessStartRanges(d);
+  }
+
+  /**
+   * Populate the editable fields with the values from the access request on component init.
+   */
+  ngOnInit(): void {
+    this.fields.update(() =>
+      this.#defaultValues().map((x, idx) => {
+        if (this.names()[idx] === 'access_starts' && new Date(x) < new Date())
+          return new Date();
+        return x;
+      }),
+    );
+
+    this.durationFormControl = this.names().map((x, idx) => {
+      if (x === 'access_starts')
+        return new FormControl('', [
+          Validators.required,
+          (control) => this.#fromDateValidator(control, idx),
+        ]);
+      else if (x === 'access_ends')
+        return new FormControl('', [
+          Validators.required,
+          (control) => this.#untilDateValidator(control, idx),
+        ]);
+      return null;
+    });
+    this.fieldErrorMessages.set(Array(this.names().length).fill(''));
+  }
+
+  /**
+   * Update the range for the until date based on the from date
+   * @param date - The from date to update the range for
+   */
+  updateAccessEndRanges(date: Date): void {
+    const newUntilMinDate = new Date(
+      date.getTime() + this.#config.accessGrantMinDays * MILLISECONDS_PER_DAY,
```

Comment:
Since our config values are given as number of days, I would use calculations based on days instead. Something like:
```js
const maxDate = new Date();
maxDate.setDate(date.getDate() + maxDays);
```
Could also avoid problems with leap seconds or DST.